packages/cpsuLambda/tests/format_1/business_states.test.ts

```diff
@@ -1,26 +1,106 @@
+    expect(businessStatus.isNothing()).toBeTruthy()
+  })
+
+  it("should convert a DispensingComplete status with Robot Collection", () => {
+    testBusinessStatusConversion("DispensingComplete", "Robot Collection", "Dispatched")
+  })
+
+  it("should handle ReadyForCollection status with Robot Collection correctly", () => {
+    testBusinessStatusConversion("ReadyForCollection", "Robot Collection", "Ready to Dispatch")
+  })
+
+  it("should convert a delivery type business status for Delivery required", () => {
+    testBusinessStatusConversion("ReadyForCollection", "Delivery required", "Ready to Dispatch")
+  })
+
+  it("should convert an item status business status for Delivery required", () => {
+    testBusinessStatusConversion("DispensingComplete", "Delivery required", "Dispatched")
+  })
 })
 
-it("should return Nothing if the item status is not in the map", () => {
-  const itemStatus: itemStatusType = "Expired"
-  const deliveryType: deliveryType = "Robot Collection"
+describe("populateTemplate function", () => {
+  let mockLogger: Logger
 
-  const businessStatus = getBusinessStatus(deliveryType, itemStatus)
-  expect(businessStatus.isNothing()).toBeTruthy()
+  beforeEach(() => {
+    mockLogger = new Logger()
+  })
+
+  function runTestCase(
+    itemStatus: itemStatusType,
+    deliveryType: deliveryType,
+    expectedBusinessStatus: string,
+    expectedTaskStatus: string
+  ) {
+    const template: string = generateTemplate({
+      MessageType: "ExampleMessageType",
+      items: [{itemID: "item1", status: itemStatus}],
+      prescriptionUUID: "123456789",
+      nHSCHI: "123456",
+      messageDate: new Date().toISOString(),
+      oDSCode: "XYZ",
+      deliveryType: deliveryType,
+      repeatNo: 1
+    })
+
+    const prescriptionItem: itemType = {
+      itemID: "item1",
+      status: itemStatus
+    }
+
+    const prescriptionDetails: requestType = {
+      MessageType: "ExampleMessageType",
+      items: [{itemID: "item1", status: itemStatus}],
+      prescriptionUUID: "123456789",
+      nHSCHI: "123456",
+      messageDate: new Date().toISOString(),
+      oDSCode: "XYZ",
+      deliveryType: deliveryType,
+      repeatNo: 1
+    }
+
+    const result = populateTemplate(template, prescriptionItem, prescriptionDetails, mockLogger)
+
+    if (result.isOk()) {
+      const entry: BundleEntry<Task> = result.value() as BundleEntry<Task>
+      expect(entry.resource!.businessStatus!.coding![0].code).toEqual(expectedBusinessStatus)
+      expect(entry.resource!.status).toEqual(expectedTaskStatus)
+    }
+  }
+
+  it("should populate template correctly for Robot Collection with ReadyForCollection status", () => {
+    runTestCase("ReadyForCollection", "Robot Collection", "Ready to Dispatch", "in-progress")
+  })
+
+  it("should populate template correctly for Robot Collection with DispensingComplete status", () => {
+    runTestCase("DispensingComplete", "Robot Collection", "Dispatched", "completed")
```

Comment:
Can we do these as one parameterised test instead of multiple tests that all just call the same function?

Reply:
The test has been parameterised.